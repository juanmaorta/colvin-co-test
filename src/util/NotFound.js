import React from 'react'
import PropTypes from 'prop-types'
import Button from '@material-ui/core/Button'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'

import styles from '../styles'

const NotFound = (props) => {
  const { classes } = props

  return (
    <div className={classes.heroUnit}>
      <div className={classes.heroContent}>
        <Typography variant='h6' align='center' color='textSecondary' paragraph>
          Estos no son los androides que busc√°is
        </Typography>
        <div className={classes.heroButtons}>
          <Grid container spacing={16} justify='center'>
            <Grid item>
              <Button variant='contained' color='primary' href='/'>Home</Button>
            </Grid>
          </Grid>
        </div>
      </div>
    </div>
  )
}

NotFound.propTypes = {
  classes: PropTypes.object.isRequired
}

export default withStyles(styles)(NotFound)
