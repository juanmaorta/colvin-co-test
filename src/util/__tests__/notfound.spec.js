/* global expect, describe, it, xit */
import React from 'react'
import { shallow } from 'enzyme'

import NotFound from '../NotFound'
import * as stubs from './stubs'

describe('NotFound snapshots', () => {
  it('should match for NotFound component', () => {
    const wrapper = shallow(
      <NotFound classes={stubs.classes} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
