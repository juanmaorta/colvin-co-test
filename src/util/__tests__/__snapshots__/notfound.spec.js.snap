// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound snapshots should match for NotFound component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(NotFound)
    classes={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "appBar": "NotFound-appBar-1",
        "blackBackGround": "NotFound-blackBackGround-5",
        "blackCardContent": "NotFound-blackCardContent-6",
        "blackCardTitle": "NotFound-blackCardTitle-3",
        "buttons": "NotFound-buttons-2",
        "card": "NotFound-card-4",
        "cardGrid": "NotFound-cardGrid-12",
        "footer": "NotFound-footer-7",
        "heroButtons": "NotFound-heroButtons-10",
        "heroContent": "NotFound-heroContent-9",
        "heroUnit": "NotFound-heroUnit-8",
        "layout": "NotFound-layout-11",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "appBar": "NotFound-appBar-1",
          "blackBackGround": "NotFound-blackBackGround-5",
          "blackCardContent": "NotFound-blackCardContent-6",
          "blackCardTitle": "NotFound-blackCardTitle-3",
          "buttons": "NotFound-buttons-2",
          "card": "NotFound-card-4",
          "cardGrid": "NotFound-cardGrid-12",
          "footer": "NotFound-footer-7",
          "heroButtons": "NotFound-heroButtons-10",
          "heroContent": "NotFound-heroContent-9",
          "heroUnit": "NotFound-heroUnit-8",
          "layout": "NotFound-layout-11",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
